### PR TITLE
[IE Samples] Export and Import of model, fixed

### PR DESCRIPTION
**Pull request by mgrdv**
Export and Import of model for speech sample [CVS-76225]

### CODE REVIEW

**Review by vladimir-dudnik — APPROVED** [review]

**Review by mryzhov — CHANGES_REQUESTED** [review]
Could you please add the pre-commit tests?

File: samples/cpp/speech_sample/main.cpp [hunk 1]
```diff
@@ -221,14 +224,18 @@ int main(int argc, char* argv[]) {
             executableNet = core.compile_model(model, deviceStr, genericPluginConfig);
         } else {
             slog::info << "Importing model to the device" << slog::endl;
-            std::istringstream streamrq(FLAGS_rg);
+            std::ifstream streamrq(FLAGS_rg, std::ios_base::binary | std::ios_base::in);
+            if (!streamrq.is_open()) {
+                throw std::runtime_error("Cannot open model file " + FLAGS_rg);
+            }
             executableNet = core.import_model(streamrq, deviceStr, genericPluginConfig);
+            streamrq.close();
```
Comment (mryzhov) [inline samples/cpp/speech_sample/main.cpp:232]: We don't need to close ifstream manually

File: samples/cpp/speech_sample/main.cpp [hunk 2]
```diff
@@ -221,14 +224,18 @@ int main(int argc, char* argv[]) {
             executableNet = core.compile_model(model, deviceStr, genericPluginConfig);
         } else {
             slog::info << "Importing model to the device" << slog::endl;
-            std::istringstream streamrq(FLAGS_rg);
+            std::ifstream streamrq(FLAGS_rg, std::ios_base::binary | std::ios_base::in);
```
Comment (mryzhov) [inline samples/cpp/speech_sample/main.cpp:227]: What if --rg is absent?

File: samples/cpp/speech_sample/main.cpp [hunk 3]
```diff
@@ -83,26 +83,29 @@ int main(int argc, char* argv[]) {
...
-        const ov::Layout tensor_layout{"NC"};
-        ov::preprocess::PrePostProcessor proc(model);
-        for (int i = 0; i < model->inputs().size(); i++) {
-            proc.input(i).tensor().set_element_type(ov::element::f32).set_layout(tensor_layout);
-        }
-        for (int i = 0; i < model->outputs().size(); i++) {
-            proc.output(i).tensor().set_element_type(ov::element::f32);
+        uint32_t batchSize = (FLAGS_cw_r > 0 || FLAGS_cw_l > 0) ? 1 : (uint32_t)FLAGS_bs;
+        std::shared_ptr<ov::Model> model;
+        if (!FLAGS_m.empty()) {
```
Comment (mryzhov) [inline samples/cpp/speech_sample/main.cpp:88]: maybe it is better to add a few comments here, e.g. preprocessing steps are not applicable for the compiled model?

Comment (mgrdv) [reply samples/cpp/speech_sample/main.cpp:88]: ok

File: samples/cpp/speech_sample/main.cpp (hunk 1 above)
Comment (mgrdv) [reply samples/cpp/speech_sample/main.cpp:232]: ok

File: samples/cpp/speech_sample/main.cpp (hunk 2 above)
Comment (mgrdv) [reply samples/cpp/speech_sample/main.cpp:227]: We have checking in function parse_and_check_command_line: "if (FLAGS_m.empty() && FLAGS_rg.empty())"

**Comment by vladimir-dudnik** [comment]
@mryzhov your comments were addressed, are you ok to merge this?

File: samples/cpp/speech_sample/main.cpp [hunk 4]
```diff
@@ -83,26 +83,30 @@ int main(int argc, char* argv[]) {
...
-        const ov::Layout tensor_layout{"NC"};
-        ov::preprocess::PrePostProcessor proc(model);
-        for (int i = 0; i < model->inputs().size(); i++) {
-            proc.input(i).tensor().set_element_type(ov::element::f32).set_layout(tensor_layout);
-        }
-        for (int i = 0; i < model->outputs().size(); i++) {
-            proc.output(i).tensor().set_element_type(ov::element::f32);
+        uint32_t batchSize = (FLAGS_cw_r > 0 || FLAGS_cw_l > 0) ? 1 : (uint32_t)FLAGS_bs;
+        std::shared_ptr<ov::Model> model;
+        // if 'm' flag is empty in command line in this case we load model using 'rg' flag
```
Comment (mryzhov) [inline samples/cpp/speech_sample/main.cpp:88]:
```suggestion
        // ------------------------------ Preprocessing ------------------------------------------------------
        // if 'm' flag is empty in command line in this case we load model using 'rg' flag
```

File: samples/cpp/speech_sample/main.cpp (hunk 3 above)
Comment (mryzhov) [reply samples/cpp/speech_sample/main.cpp:89]: Thanks, but this is a sample and we have to explain why we are skipping preprocessing steps for some cases, what about the following comment:
"The preprocessing steps can be done only for loaded network and are not applicable for the imported network (already compiled)?

File: samples/cpp/speech_sample/main.cpp (hunk 4 above)
Comment (mryzhov) [inline samples/cpp/speech_sample/main.cpp:88]:
```suggestion
        // the preprocessing steps can be done only for loaded network and are not applicable for the imported network (already compiled)
```

**Comment by mryzhov** [comment]
> @mryzhov your comments were addressed, are you ok to merge this?
I'm expecting the pre-commit tests to avoid such bugs in the future. Have we enabled the pre-commit tests for this functionality?

**Comment by dorloff** [comment]
I agree with @mryzhov that we cannot leave the current issues with tests unresolved, especially with huge changes in API. We have already faced a number of issues and may have more without testing.

**Pull request merged**